db/model/Gdoc/archieToGdoc.ts

```diff
@@ -0,0 +1,241 @@
+    yield* propertyToArchieMLString("supertitle", article)
+    yield* propertyToArchieMLString("template", article)
+    yield* propertyToArchieMLString("byline", article)
+    yield* propertyToArchieMLString("dateline", article)
+    yield* propertyToArchieMLString("excerpt", article)
+    // TODO: inline refs
+    yieldMultiBlockPropertyIfDefined("summary", article, article.summary)
+    yieldMultiBlockPropertyIfDefined("citation", article, article.summary)
+    yield* propertyToArchieMLString("cover-image", article)
+    yield* propertyToArchieMLString("cover-color", article)
+    yield* propertyToArchieMLString("featured-image", article)
+    yield ""
+    if (article.body) {
+        yield "[+body]"
+        for (const block of article.body) {
+            const rawBlock = enrichedBlockToRawBlock(block)
+            const lines = [
+                ...owidRawArticleBlockToArchieMLStringGenerator(rawBlock),
+            ]
+            yield* lines
+            yield ""
+        }
+        yield "[]"
+    }
+}
+
+interface TextFragment {
+    text: string
+    style: docs_v1.Schema$TextStyle
+}
+
+interface Line {
+    fragments: TextFragment[]
+}
+
+const defaultStyle: docs_v1.Schema$TextStyle = {
+    bold: false,
+    italic: false,
+    underline: false,
+    strikethrough: false,
+    baselineOffset: "NONE",
+    link: undefined,
+}
+function mergeStyleStack(styleStack: docs_v1.Schema$TextStyle[]) {
+    const mergedStyle = styleStack.reduce(
+        (acc, style) => ({
+            ...acc,
+            ...style,
+        }),
+        defaultStyle
+    )
+    return mergedStyle
+}
+
+function* convertCheerioNodesToTextFragments(
+    nodes: Iterable<CheerioElement>,
+    parentStyleStack: docs_v1.Schema$TextStyle[]
+): Generator<TextFragment, void, undefined> {
+    for (const node of nodes) {
+        if (node.type === "text") {
+            const currentStyle = mergeStyleStack(parentStyleStack)
+            const text = node.data
+            if (text && text !== "") yield { text, style: currentStyle }
+        } else if (node.type === "tag") {
+            const tag = node.name
+            const styleStack = [...parentStyleStack]
+            if (tag === "b") {
+                styleStack.push({ bold: true })
+            } else if (tag === "i") {
+                styleStack.push({ italic: true })
+            } else if (tag === "q") {
+                styleStack.push({ italic: true })
+            } else if (tag === "u") {
+                styleStack.push({ underline: true })
+            } else if (tag === "s") {
+                styleStack.push({ strikethrough: true })
+            } else if (tag === "sup") {
+                styleStack.push({ baselineOffset: "SUPERSCRIPT" })
+            } else if (tag === "sub") {
+                styleStack.push({ baselineOffset: "SUBSCRIPT" })
+            } else if (tag === "a") {
+                // TODO: ref links have a class assigned and will need to be handled here
```

Comment:
Does this need to be done in this PR?

Reply:
I'd rather work on this in a separate PR but will tackle this soon.